js_runner.go

```diff
@@ -66,13 +68,39 @@ func (runner *JSRunner) Init(funcSource string) error {
 		return otto.UndefinedValue()
 	})
 
+	runner.js.Set("console", map[string]interface{}{
+		"error": func(s string) { errorMessageCallback(s)},
+		"log": func(s string) { logMessageCallback(s)},
+	})
+
 	if _, err := runner.SetFunction(funcSource); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+var logMessageCallback = defaultLogFunction
+var errorMessageCallback = defaultLogFunction
+
+var logCallbackLock sync.Mutex
+
+func (runner *JSRunner) SetErrorCallback(logFn func(string)){
+	logCallbackLock.Lock()
+	errorMessageCallback = logFn
+	logCallbackLock.Unlock()
+}
+
+func (runner *JSRunner) SetLogCallback(logFn func(string)){
```

Comment:
If we do the above change, for consistency, let's change this one too.

```suggestion
func (runner *JSRunner) SetConsoleLogCallback(logFn func(string)){
```